code/datums/components/relay_attackers.dm

```diff
@@ -1,25 +1,24 @@
 /**
- * This element registers to a shitload of signals which can signify "someone attacked me".
+ * This component registers to a shitload of signals which can signify "someone attacked me".
+ * While this holds no state, it's a component because elements have no source control and it can be added or removed by a bunch of other shit.
  * If anyone does it sends a single "someone attacked me" signal containing details about who done it.
  * This prevents other components and elements from having to register to the same list of a million signals, should be more maintainable in one place.
  */
-/datum/element/relay_attackers
+/datum/component/relay_attackers
+	dupe_mode = COMPONENT_DUPE_SOURCES
```

Comment:
I understand why this is being done but also, why is this being done? 

Is there a reason why we must remove the element/component when we're not using using it? All it does is send a signal. If there's no signal handler it does effectively nothing.

Reply:
I'll be honest it's more about the fact that applying an element multiple times will try to reregister the same signals
I could make it add a trait and not reregister if it already has the trait but that seems very messy

Reply:
Well it already adds a trait, `TRAIT_RELAYING_ATTACKER`, so it's not like it's... much worse I guess